src/SOS/lldbplugin/services.cpp

```diff
@@ -1401,14 +1401,6 @@ LLDBServices::GetCurrentThreadId(
         return E_FAIL;
     }
 
-    // This is allow the a valid current TID to be returned to 
-    // workaround a bug in lldb on core dumps.
-    if (g_currentThreadIndex != (ULONG)-1)
-    {
-        *id = g_currentThreadIndex;
-        return S_OK;
-    }
-
```

Comment:
Don't we want to keep this for system generated core dumps?

Reply:
System generated core dumps are still supported. `setsostid` will still work for system generated core dumps. The command is better now because more than one mapping can be added.